sdx-submitter.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/streadway/amqp"
+	"io/ioutil"
+	"os"
+)
+
+func main() {
+	// Read a source file and place it on a rabbit topic exchange
+	// Exchange , Queue and binding must be in place before use
+
+	var url, exchange, routingKey, encryptionKey, signingKey, name, password string
+	var port, host, queue, vhost, messageFilePath string
+	var msgBody []byte
+	var err error
+	const PRINT_MSG_CHAR_COUNT = 20 // only print the first few characters of the message
+
+	// access command line parameters
+
+	flag.StringVar(&name, "n", "", "name of the rabbit user")
+	flag.StringVar(&password, "p", "", "password of the rabbit user")
+	flag.StringVar(&port, "rport", "", "port used to connect to rabbit")
+	flag.StringVar(&host, "rhost", "", "hostname used to connect to rabbit")
+	flag.StringVar(&vhost, "v", "", "vhostname used to connect to rabbit")
+	flag.StringVar(&url, "u", "", "url connection string ")
+	flag.StringVar(&queue, "q", "", "name of the rabbit queue")
+	flag.StringVar(&exchange, "x", "", "name of the rabbit exchange")
+	flag.StringVar(&routingKey, "r", "", "rabbit routing key")
+	flag.StringVar(&encryptionKey, "e", "", "path to a private key file used for encryption")
+	flag.StringVar(&signingKey, "s", "", "path to a private key used for signing")
+	flag.StringVar(&messageFilePath, "f", "", "path to filename to send")
+
+	flag.Parse()
+
+	// If no value given on command line , then look at environment variables , else use a default value
+
+	name = getFromEnvIfEmpty(name, "RABBITMQ_DEFAULT_USER", "guest")
+	password = getFromEnvIfEmpty(password, "RABBITMQ_DEFAULT_PASS", "guest")
+	port = getFromEnvIfEmpty(port, "RABBITMQ_PORT", "5672")
+	host = getFromEnvIfEmpty(host, "RABBITMQ_HOST", "localhost")
+	queue = getFromEnvIfEmpty(queue, "RABBIT_SURVEY_QUEUE", "rabbit")
+	exchange = getFromEnvIfEmpty(exchange, "RABBITMQ_EXCHANGE", "message")
+	vhost = getFromEnvIfEmpty(vhost, "RABBITMQ_DEFAULT_VHOST", "%2f")
+
+	if url == "" {
+		url = fmt.Sprintf("amqp://%s:%s@%s:%s/%s", name, password, host, port, vhost)
+	} else {
+		fmt.Println("url use overrides specific parameters") //A specific URL is set , so use it
+	}
+
+	msgBody, err = getBody(messageFilePath)
+	failOnError(err, "could not read message body")
+
+	// If encyrpt specified then encrypt
+	// if sign specified then sign
+
+	err = sendToRabbit(url, exchange, queue, routingKey, msgBody)
+	failOnError(err, "unable to send message to rabbitmq")
+
+	var msgSize = len(msgBody)
+	if msgSize < PRINT_MSG_CHAR_COUNT {
+		fmt.Println(fmt.Sprintf("message:'%s' (len=%d) published to exchange:'%s' using routing key:'%s'", string(msgBody), msgSize, exchange, routingKey))
+	} else {
+		fmt.Println(fmt.Sprintf("message:'%s...' (len=%d) published to exchange:'%s' using routing key:'%s'", string(msgBody[0:PRINT_MSG_CHAR_COUNT]), msgSize, exchange, routingKey))
+	}
```

Comment:
Wouldn't print out the actual message as it's likely to be pretty long (especially if it's the encrypted version that's actually published).

Reply:
It prints up to the first 20 chars of a message only , e.g 
message:'Something Something ...' (len=27) published to exchange:'message' using routing key:''